### PR TITLE
Enhancement: Add methods go get/set modelless form data.

### PR DESCRIPTION
**Pull request by ADmad**
This change will avoid having to modify request data to populate modelless form with initial values. I'll finish up the PR if the idea is approved.

Refs #12155.

### CODE REVIEW

File: src/Form/Form.php [hunk 1]
```diff
@@ -304,6 +304,26 @@ protected function _execute(array $data)
...
+    public function getData($field = null)
+    {
+        if ($field === null) {
+            return $this->_data;
+        }
+
+        return Hash::get($this->_data, $field);
+    }
+
+    public function setData(array $data, $merge = true)
```
[the remark this answers is not in the document]
Comment (markstory) [reply src/Form/Form.php]: Should overwriting be a different method? Having bool flags is sometimes a code smell.

[the remark this answers is not in the document]
Comment (ADmad) [reply src/Form/Form.php]: In that case I would just remove the second param and always overwrite. I don't see many use cases for merging data. Merging would still be possible using `setData($data + $form->getData())` on the off chance it's needed.

[the remark this answers is not in the document]
Comment (markstory) [reply src/Form/Form.php]: I'm good with not having a merge until we need it.

**Comment by ADmad** [comment]
Updated as per feedback and tests added.

**Pull request merged**